saged/datasets.py

```diff
@@ -231,3 +244,443 @@ def get_all_data(self) -> np.array:
         X: The gene expression data in a genes x samples array
         """
         raise NotImplementedError
+
+
+class RefineBioLabeledDataset(LabeledDataset):
+    """ A dataset designed to store labeled data from a refine.bio compendium """
+    def __init__(self,
+                 expression_df: pd.DataFrame,
+                 sample_to_label: Dict[str, str],
+                 sample_to_study: Dict[str, str],
+                 ):
+        """
+        An initializer for the RefineBioLabeledDataset. Typically this initializer
+        will not be used directly, but will instead be called by
+        RefineBioLabeledDataset.from_paths
+
+        Arguments
+        ---------
+        expression_df: The dataframe containing expression data where rows are genes and
+            columns are samples
+        sample_to_label: A mapping between sample accessions and their phenotype labels
+        sample_to_study: A mapping between sample accessions and their study accessions
+        """
+        self.all_expression = expression_df
+        self.sample_to_label = sample_to_label
+        self.sample_to_study = sample_to_study
+
+        # We will handle subsetting by creating views of the full dataset.
+        # Most functions will access current_expression instead of all_expression.
+        self.current_expression = expression_df
```

Comment:
Do you want to set `self.data_changed = False` here? Or does it not matter?

Reply:
If `self.data_changed` were set to False in the init function, it would throw an error when we try to `get_studies()` before something else sets self.data_chaged to True. I could set `self.data_chaged = True` in the init function, but that seems a little confusing. I'll stick with checking whether the attribute `data_changed` is present, but I'll add a comment explaining why that's necessary